### PR TITLE
front: integrate ui-core select component in stdcm

### LINKED ISSUE #8820: Use the new Select for Speed Limit Tag selector
Replace 
image
 By
image

### PR DESCRIPTION
**Pull request by achrafmohye**
Closes #8820

### CODE REVIEW

File: front/tests/pages/scenario-page-model.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ class ScenarioPage extends BasePage {
     this.getScenarioDeleteConfirmBtn = page
       .locator('#modal-content')
       .getByText('Supprimer', { exact: true });
-    this.getSpeedLimitSelector = page.getByTestId('speed-limit-by-tag-selector');
+    this.getSpeedLimitSelector = page.locator('speed-limit-by-tag-selector');
     this.getItineraryModule = page.getByTestId('itinerary');
```
Comment (achrafmohye) [inline front/tests/pages/scenario-page-model.ts:102]: Taking a look on the tests 🧐

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 2]
```diff
@@ -39,11 +39,10 @@ export default function SpeedLimitByTagSelector({
       >
         <img width="32px" src={icon} alt="speedometer" />
         <span className="text-muted">{t('speedLimitByTag')}</span>
-        <SelectImprovedSNCF
-          sm
+        <Select
+          small
```
Comment (SharglutDev) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: In the mockup, the component seems to have the same font size as the rest of the form

```suggestion
```

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 3]
```diff
@@ -53,6 +52,8 @@ export default function SpeedLimitByTagSelector({
               dispatchUpdateSpeedLimitByTag(null);
             }
           }}
+          getOptionLabel={(option) => option}
+          getOptionValue={(option) => option}
```
Comment (SharglutDev) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: Still find pretty weird that we will have to do this each time we use this component

Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: Would seem to agree, but we have to accept an array of strings for the suggestions or have a specific format for them !

Comment (kmer2016) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: The purpose of these two props is to allow users to pass their data in its raw form without needing to transform it beforehand. You provide the data as-is and pass two callbacks to specify how to display and identify each element in the list. The benefit becomes more apparent when dealing with more complex data than simple strings.

Maybe we could make these props optional if the list of select options consists of strings?

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 4]
```diff
@@ -39,11 +39,10 @@ export default function SpeedLimitByTagSelector({
...
         <span className="text-muted">{t('speedLimitByTag')}</span>
-        <SelectImprovedSNCF
-          sm
+        <Select
+          small
           disabled={disabled}
-          withSearch
-          dataTestId="speed-limit-by-tag-selector"
+          id="speed-limit-by-tag-selector"
           value={speedLimitByTag || t('noSpeedLimitByTag').toString()}
```
Comment (emersion) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: Would be nicer to not use strings intended for humans (such as translations) as `<select>` values (which are meant to be unique and are not displayed to the user). A simple solution is to move `|| t('noSpeedLimitByTag').toString()` to `getOptionLabel()`. Then we can get rid of the `e !== t('noSpeedLimitByTag')` check.

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 3 above)
Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: I agree that allowing `<Select>` users to pass their own data type is worthwhile, and I also agree that for many cases it's cumbersome to pass trivial `getOptionLabel`/`getOptionValue` callbacks. I don't recommend adding more knobs to `<Select>` props because it would increase the complexity of its API. Instead, a separate helper function could be introduced. Here's one of the possible solutions:

```tsx
type SelectFixedOption = {
  label: string;
  value: string;
};

const withFixedOptions = (options: SelectFixedOption[]) => {
  return {
    options,
    getOptionLabel: (option) => option.label,
    getOptionValue: (option) => option.value,
  };
};

// usage:
const options = speedLimitsTagsList.map((tag) => ({ label: tag, value: tag }));
<Select {...withFixedOptions(options)} />
```

Many variations of this are possible as well, e.g. only providing `getOptionLabel`/`getOptionValue` helpers for `string[]` (but wouldn't work if you want to use a label different from the value, which is quite common):

```tsx
const withStringArray = (options: string[]) => {
  return {
    options,
    getOptionLabel: (option) => option,
    getOptionValue: (option) => option,
  };
};

// usage:
<Select {...withStringArray(speedLimitsTagsList)} />
```

or replacing `SelectOption.value` with an `onSelect` callback:

```tsx
type SelectFixedOption = {
  label: string;
  onSelect: () => void;
};

const withFixedOptions = (options: SelectFixedOption[]) => {
  const optionsMap = new Map(options.map((option, index) => [option, index.toString()]));
  return {
    options,
    getOptionLabel: (option) => option.label,
    getOptionValue: (option) => optionsMap.get(option),
    onChange: (option) => option?.onSelect(),
  };
};

// usage:
const options = speedLimitsTagsList.map((tag) =>
  ({ label: tag, onSelect: () => dispatchUpdateSpeedLimitByTag(tag) })
);
<Select {...withFixedOptions(options)} />
```

Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: TBH… I'm not even sure why `<Select>` requires callers to pass a `value` at all. I think we could just auto-generate a `value` for each option based on their index. We wouldn't loose the current selection on component re-renders (e.g. when the option list changes) because internally `<Select>` does `value={selectedOption ? getOptionValue(selectedOption) : undefined}`.

Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: I now realize that if we want to drop `value`, we wouldn't even need to generate them based on indexes: we can just grab the selected index via [`selectedIndex`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedIndex).

Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: i like the solution of having `withFixedOptions` but if have to change the internal functioning of the `Select`   component we can create an issue in` ui-core`

Comment (kmer2016) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: > I agree that allowing `<Select>` users to pass their own data type is worthwhile, and I also agree that for many cases it's cumbersome to pass trivial `getOptionLabel`/`getOptionValue` callbacks. I don't recommend adding more knobs to `<Select>` props because it would increase the complexity of its API. Instead, a separate helper function could be introduced. Here's one of the possible solutions:
> 
> ```tsx
> type SelectFixedOption = {
>   label: string;
>   value: string;
> };
> 
> const withFixedOptions = (options: SelectFixedOption[]) => {
>   return {
>     options,
>     getOptionLabel: (option) => option.label,
>     getOptionValue: (option) => option.value,
>   };
> };
> 
> // usage:
> const options = speedLimitsTagsList.map((tag) => ({ label: tag, value: tag }));
> <Select {...withFixedOptions(options)} />
> ```
> 
> Many variations of this are possible as well, e.g. only providing `getOptionLabel`/`getOptionValue` helpers for `string[]` (but wouldn't work if you want to use a label different from the value, which is quite common):
> 
> ```tsx
> const withStringArray = (options: string[]) => {
>   return {
>     options,
>     getOptionLabel: (option) => option,
>     getOptionValue: (option) => option,
>   };
> };
> 
> // usage:
> <Select {...withStringArray(speedLimitsTagsList)} />
> ```
> 
> or replacing `SelectOption.value` with an `onSelect` callback:
> 
> ```tsx
> type SelectFixedOption = {
>   label: string;
>   onSelect: () => void;
> };
> 
> const withFixedOptions = (options: SelectFixedOption[]) => {
>   const optionsMap = new Map(options.map((option, index) => [option, index.toString()]));
>   return {
>     options,
>     getOptionLabel: (option) => option.label,
>     getOptionValue: (option) => optionsMap.get(option),
>     onChange: (option) => option?.onSelect(),
>   };
> };
> 
> // usage:
> const options = speedLimitsTagsList.map((tag) =>
>   ({ label: tag, onSelect: () => dispatchUpdateSpeedLimitByTag(tag) })
> );
> <Select {...withFixedOptions(options)} />
> ```

I totally agree with you. Thank you for your helpful feedback and examples!

Comment (kmer2016) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: > I now realize that if we want to drop `value`, we wouldn't even need to generate them based on indexes: we can just grab the selected index via [`selectedIndex`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedIndex).

I see your point about auto-generating values based on the index. However, I believe keeping the value prop is important. Without it, we wouldn’t be able to initialize the Select component with a value from a Redux store. This makes it harder to set the correct selected option when loading from the store.
Do you have any suggestions on how to handle this situation with your proposal?

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 4 above)
Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: @emersion can you check the fix and if it's done properly

File: front/tests/005-operational-studies.spec.ts [hunk 5]
```diff
@@ -93,15 +93,6 @@ test.describe('Testing if all mandatory elements simulation configuration are lo
 
     await scenarioPage.checkPathfindingDistance('34.000 km');
 
-    // TODO: move this test in his own file
```
Comment (achrafmohye) [inline front/tests/005-operational-studies.spec.ts:77]: @Maymanaf like discussed can you check the test part :)

Comment (Maymanaf) [reply front/tests/005-operational-studies.spec.ts:77]: This part of the tests covers the old component and is no longer needed, aas we will have two tests that address the new behavior:

- A new test: https://github.com/OpenRailAssociation/osrd/issues/8647
- A refactor of test 005: https://github.com/OpenRailAssociation/osrd/issues/8648

**Review by Maymanaf — APPROVED** [review]

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 6]
```diff
@@ -39,20 +41,14 @@ export default function SpeedLimitByTagSelector({
...
+          value={speedLimitByTag || ''}
           onChange={(e) => {
-            if (e && e !== t('noSpeedLimitByTag')) {
-              dispatchUpdateSpeedLimitByTag(e);
-            } else {
-              dispatchUpdateSpeedLimitByTag(null);
-            }
+            dispatchUpdateSpeedLimitByTag(e || null);
           }}
+          {...createStringSelectOptions(speedLimitsTagsList, t('noSpeedLimitByTag').toString())}
```
Comment (SharglutDev) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: Don't think you need to do that, the default one is already added in the list in `useStoreDataForSpeedLimitByTagSelector`

```suggestion
          {...createStringSelectOptions(speedLimitsTagsList)}
```

File: front/src/utils/uiCoreHelpers.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+// eslint-disable-next-line import/prefer-default-export
+export const createStringSelectOptions = (opts: string[], labelPlaceholder?: string) => ({
```
Comment (SharglutDev) [inline front/src/utils/uiCoreHelpers.ts]: Maybe a comment of why we need to have this util :)

Comment (SharglutDev) [inline front/src/utils/uiCoreHelpers.ts]: Should the function be more generic in case `options` are a bit more complex ?

Comment (achrafmohye) [reply front/src/utils/uiCoreHelpers.ts]: I can introduce a second function for a usage of non string cases where the options are objects ?

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 6 above)
Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: Done

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 8]
```diff
@@ -30,29 +30,23 @@ export default function SpeedLimitByTagSelector({
...
       <div
         className={cx('osrd-config-item-container', {
           'd-flex align-items-center gap-10': condensed,
         })}
       >
-        <img width="32px" src={icon} alt="speedometer" />
-        <span className="text-muted">{t('speedLimitByTag')}</span>
-        <SelectImprovedSNCF
-          sm
+        <Select
```
Comment (SharglutDev) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:41]: This will also change the UI for the selector in operational studies when adding/editing a train, resulting to a "zombified" interface. Do we have the ok from PO's or @thibautsailly ?
Moreover, in operational studies, the label will dislay "Code compo" instead of "Code de composition". Is it a change we want ?

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 9]
```diff
@@ -30,29 +30,23 @@ export default function SpeedLimitByTagSelector({
...
+          id="speed-limit-by-tag-selector"
+          value={speedLimitByTag || ''}
+          label={t('speedLimitByTagAbbrev')}
           onChange={(e) => {
-            if (e && e !== t('noSpeedLimitByTag')) {
-              dispatchUpdateSpeedLimitByTag(e);
-            } else {
-              dispatchUpdateSpeedLimitByTag(null);
-            }
+            dispatchUpdateSpeedLimitByTag(e || null);
```
Comment (SharglutDev) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: We no longer filter the `noSpeedLimitByTag` value, meaning when we select it, the store is updated with that and when adding/editing the train, we send this non consistent value instead of null. I'm afraid it might result to some weird things in the simulation. Maybe we should leave it like it was before ?

File: front/src/utils/uiCoreHelpers.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+      id: string;
+    }
+  | undefined;
+
+// Utility to create standardized options for <Select> components.
+// `getOptionLabel`: Displays the option or a placeholder if empty.
+// `getOptionValue`: Uses the option itself as the value.
+export const createStringSelectOptions = (opts: string[]) => ({
+  options: opts,
+  getOptionLabel: (option: string) => option || '',
```
Comment (SharglutDev) [inline front/src/utils/uiCoreHelpers.ts]: Don't think we need that

```suggestion
  getOptionLabel: (option: string) => option,
```

File: front/src/applications/stdcmV2/components/StdcmOpSchedule.tsx [hunk 11]
```diff
@@ -95,9 +96,8 @@ const StdcmOpSchedule = ({
               onArrivalTypeChange(e as ArrivalTimeTypes);
             }
           }}
-          options={['preciseTime', 'asSoonAsPossible']}
+          {...createStringSelectOptions(['preciseTime', 'asSoonAsPossible'])}
```
Comment (SharglutDev) [inline front/src/applications/stdcmV2/components/StdcmOpSchedule.tsx:91]: Wasn't the point of using `createStringSelectOptions` not to have to pass `getOptionLabel` ?

File: front/src/utils/uiCoreHelpers.ts (hunk 7 above)
Comment (SharglutDev) [reply front/src/utils/uiCoreHelpers.ts]: Would have been better to be able to have one function but it seems complicated

File: front/src/applications/stdcmV2/components/StdcmOpSchedule.tsx (hunk 11 above)
Comment (achrafmohye) [reply front/src/applications/stdcmV2/components/StdcmOpSchedule.tsx:91]: Yes but it's a special case where we have a translation as label don't think that we want a function for each case ?

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 8 above)
Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:41]: It's the label suggested by @thibautsailly in the sketch, going to ask him if it's the the good label to have

Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:41]: Here is the respone of @thibautsailly `To respect the width of the component, we don't really have a choice but to shorten the term, so it's indeed what is on the mockup that needs to be written, in English, no need to translate, you can leave it as 'code compo.'`

File: front/src/applications/stdcmV2/components/StdcmOpSchedule.tsx (hunk 11 above)
Comment (SharglutDev) [reply front/src/applications/stdcmV2/components/StdcmOpSchedule.tsx:91]: Hmm ok :)

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 8 above)
Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:41]: @thibautsailly is ok with having ui-core component in Operational studies, showed him a before and after screens of the component used simulation settings to make sure :)

Comment (SharglutDev) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:41]: Ok for @thibautsailly

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 3 above)
Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: > we wouldn’t be able to initialize the Select component with a value from a Redux store

What's in the Redux store is completely orthogonal to the props passed to `Select` I believe. The current selected option in `<Select value=…>` is of type `T`, it's _not_ a `string` compared to `getOptionValue`.

In other words, I don't think dropping `getOptionValue` in `Select` would make any difference: in OSRD we would just drop it and everything would still continue to work as before.

See https://github.com/OpenRailAssociation/osrd-ui/pull/538

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx [hunk 12]
```diff
@@ -30,29 +30,27 @@ export default function SpeedLimitByTagSelector({
...
       >
-        <img width="32px" src={icon} alt="speedometer" />
-        <span className="text-muted">{t('speedLimitByTag')}</span>
-        <SelectImprovedSNCF
-          sm
+        <Select
           disabled={disabled}
-          withSearch
-          dataTestId="speed-limit-by-tag-selector"
-          value={speedLimitByTag || t('noSpeedLimitByTag').toString()}
```
Comment (emersion) [inline front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:47]: I think we loose the ability to clear the speed limit by tag selection?

Could be fixed with something like `placeholder={t('noSpeedLimitByTag')}`.

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 4 above)
Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx]: Yeah, sounds good!

File: front/src/utils/uiCoreHelpers.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
+type SelectFixedOption =
+  | {
+      label: string;
+      id: string;
+    }
+  | undefined;
```
Comment (emersion) [inline front/src/utils/uiCoreHelpers.ts:7]: Why allow `undefined` here?

File: front/src/utils/uiCoreHelpers.ts [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  getOptionValue: (option: string) => option,
+});
+
+// Utility function for handling fixed label-value object arrays
+// structure may vary, and custom label/value mappings are needed.
+export const createFixedSelectOptions = (opts: SelectFixedOption[]) => ({
+  options: opts,
+  getOptionLabel: (option: SelectFixedOption) => option?.label ?? '',
+  getOptionValue: (option: SelectFixedOption) => option?.id ?? '',
+});
```
Comment (emersion) [inline front/src/utils/uiCoreHelpers.ts:24]: It would probably be best to have these helpers in osrd-ui?

File: front/src/utils/uiCoreHelpers.ts (hunk 13 above)
Comment (achrafmohye) [reply front/src/utils/uiCoreHelpers.ts:7]: To accommodate the the `value` and `options` of the select in `StdcmOperationalPoint` since the value can be undefined

File: front/src/utils/uiCoreHelpers.ts (hunk 14 above)
Comment (achrafmohye) [reply front/src/utils/uiCoreHelpers.ts:24]: Had this discussion with @kmer2016, we talked about having a default behaviour at least for the case where the `options` are simple strings and for complex types keep the getOptionLabel optional used when needed

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 12 above)
Comment (achrafmohye) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:47]: a placeholder can be a good enhancement

File: front/src/utils/uiCoreHelpers.ts (hunk 13 above)
Comment (achrafmohye) [reply front/src/utils/uiCoreHelpers.ts:7]: in matter fact the default value of the `Select` is already added in `useStoreDataForSpeedLimitByTagSelector` so i don't think it is necessary since it's the first value of the options passed to the `Select`

Comment (achrafmohye) [reply front/src/utils/uiCoreHelpers.ts:7]: @emersion what do you think ?

**Review by clarani — DISMISSED** [review]
- can you rebase so we can see the new inputs added by Sarah ?
- the inputs are not aligned
image

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 12 above)
Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:47]: I mean, we're loosing functionality here, it's a regression, no?

File: front/src/utils/uiCoreHelpers.ts (hunk 14 above)
Comment (emersion) [reply front/src/utils/uiCoreHelpers.ts:24]: Not sure I understand, but it sounds a bit too automagical/surprising to me…

File: front/src/utils/uiCoreHelpers.ts (hunk 13 above)
Comment (emersion) [reply front/src/utils/uiCoreHelpers.ts:7]: `sortedChOptions` can never contain an `undefined` item. Sorry, I don't understand the justification.

`undefined` items don't make sense to me, because `Select` has a separate mechanism to unset the selected option: `placeholder`.

File: front/src/utils/uiCoreHelpers.ts (hunk 14 above)
Comment (kmer2016) [reply front/src/utils/uiCoreHelpers.ts:24]: I suggest we keep the `utils` in osrd for now, and I'll prepare an issue on the osrd-ui side to move them later. Does that sound good to you?

Comment (emersion) [reply front/src/utils/uiCoreHelpers.ts:24]: Sure!

File: front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx (hunk 12 above)
Comment (emersion) [reply front/src/common/SpeedLimitByTagSelector/SpeedLimitByTagSelector.tsx:47]: To expand on this a bit: before a user could select a speed limit by tag and then unselect it. Now there is no way to unselect it. Once selected, the user is stuck with the speed limit by tag.

File: front/src/utils/uiCoreHelpers.ts (hunk 14 above)
Comment (kmer2016) [reply front/src/utils/uiCoreHelpers.ts:24]: ok, here the [issue](https://github.com/OpenRailAssociation/osrd-ui/issues/555)

File: front/src/utils/uiCoreHelpers.ts (hunk 13 above)
Comment (emersion) [reply front/src/utils/uiCoreHelpers.ts:7]: Actually that's not possible because of an osrd-ui bug: https://github.com/OpenRailAssociation/osrd-ui/pull/557

**Review by kmer2016 — APPROVED** [review]
LGTM !

**Comment by emersion** [comment]
Can you squash the commits together?

**Comment by emersion** [comment]
Ow wait, is it intentional that this PR changes the playwright config?

**Comment by emersion** [comment]
diff also contains other unrelated changes, such as ESLint commands to disable some lints. However I don't see why this PR touches these unrelated parts of the code and why we should add these when the lints are already globally disabled in `.eslintrc`…

File: front/src/common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector.ts [hunk 15]
```diff
@@ -43,12 +41,14 @@ export const useStoreDataForSpeedLimitByTagSelector = () => {
     }
   }, [error]);
 
-  let speedLimitsByTags = compact(uniq(concat(speedLimitTags, speedLimitsTagsByInfraId)));
+  const speedLimitsByTags = compact(uniq(concat(speedLimitTags, speedLimitsTagsByInfraId)));
   const speedLimitsByTagsOrdered = useMemo(() => speedLimitsByTags.sort(), [speedLimitsByTags]);
```
Comment (emersion) [inline front/src/common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector.ts:45]: Random drive-by comment: this `useMemo` never memorizes anything because its input changes at each render.

(But that's unrelated to this PR.)

**Comment by emersion** [comment]
The two comments above about unrelated changes making their way into this PR are now resolved.

**Review by Maymanaf — APPROVED** [review]
LGTM for e2e tests

**Review by emersion — APPROVED** [review]
LGTM (not tested)
